src/rust/lib/enso-prelude/src/rc.rs

```diff
@@ -0,0 +1,25 @@
+//! This module defines utilities for working with the `Rc` and `Weak` types.
+
+use std::rc::Rc;
+use std::rc::Weak;
+use super::option::*;
+
+
+/// Mapping methods to the `Weak` type.
+pub trait WeakOps {
+    type Target;
+    fn for_each     <U,F> (self  , f:F) where F : FnOnce(Self::Target)  -> U;
+    fn for_each_ref <U,F> (&self , f:F) where F : FnOnce(&Self::Target) -> U;
+}
```

Comment:
An idea: maybe we could just implement OpitonOps for Weak<T> and have all those methods at once?

Reply:
Hmmm, interesting!

Reply:
I added comment to think about it. Its not very simple as the types will get really complex. Styill, interesting idea.